torch/fx/experimental/symbolic_shapes.py

```diff
@@ -1965,6 +1969,9 @@ def evaluate_expr(self, expr: "sympy.Expr", hint=None):
         else:
             concrete_val = sympy.sympify(hint)
 
+        if self.frozen:
+            log.warning(f"Ignored guard {expr} == {concrete_val}, this could result in accuracy problems")
```

Comment:
now that you point this whole flow out, why do we only warn here instead of error?

would we accumulate the guard into a subsequent recompile or would it always get dropped on the floor?

Reply:
I am afraid this will cause test failures and want the warning first. I'm testing this as error in next PR

Reply:
testing on a stack that removes some candidates for misbehavior: https://github.com/pytorch/pytorch/pull/99440/files